src/servers/src/server.rs

```diff
@@ -32,20 +32,31 @@ pub(crate) type AbortableStream = Abortable<TcpListenerStream>;
 pub type ServerHandler = (Box<dyn Server>, SocketAddr);
 
 /// [ServerHandlers] is used to manage the lifecycle of all the services like http or grpc in the GreptimeDB server.
-#[derive(Clone, Default)]
-pub struct ServerHandlers {
-    handlers: Arc<RwLock<HashMap<String, ServerHandler>>>,
+#[derive(Clone)]
+pub enum ServerHandlers {
+    Init(Arc<std::sync::Mutex<HashMap<String, ServerHandler>>>),
+    Started(Arc<RwLock<HashMap<String, Box<dyn Server>>>>),
```

Comment:
By the way, is it going to change after server started? Can we remove the lock here?